.github/workflows/main.yaml

```diff
@@ -10,22 +10,20 @@ jobs:
 
     steps:
       - name: Checkout
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
 
       - name: Use Node ${{ matrix.node }}
-        uses: actions/setup-node@v1
+        uses: actions/setup-node@v3
         with:
           node-version: ${{ matrix.node }}.x
 
       - name: Use cached node_modules
-        uses: actions/cache@v1
+        uses: actions/cache@v3
         with:
           path: node_modules
           key: nodeModules-${{ matrix.node }}-${{ hashFiles('**/package-lock.json') }}
-          restore-keys: |
-            nodeModules-
```

Comment:
removed these to avoid restoring the `node_modules` cache for other versions of Node.js - this can sometimes lead to weird build errors when native modules need to be built differently for different versions of Node.js